swim/join_sender.go

```diff
@@ -229,6 +231,10 @@ func (j *joinSender) JoinCluster() ([]string, error) {
 
 	for {
 		if j.node.Destroyed() {
+			j.node.emit(JoinFailedEvent{
```

Comment:
to emit events from the join sender I had a look what was already available and found: https://github.com/uber/ringpop-go/blob/master/swim/join_sender.go#L278 .

I'd rather pass in an event emitter here to be more verbose about emitting events, but for that I would like more clarity around the event emitter pattern that is used in ringpop go to see if we want to keep it and if we need to improve on this.